overlay/ports/ffmpeg/portfile.cmake

```
@@ -0,0 +1,1081 @@
+vcpkg_from_github(
+    OUT_SOURCE_PATH SOURCE_PATH
+    REPO ffmpeg/ffmpeg
+    REF "n${VERSION}"
+    SHA512 3b273769ef1a1b63aed0691eef317a760f8c83b1d0e1c232b67bbee26db60b4864aafbc88df0e86d6bebf07185bbd057f33e2d5258fde6d97763b9994cd48b6f
+    HEAD_REF master
+    PATCHES
+        0001-create-lib-libraries.patch
+        0002-fix-msvc-link.patch
+        0003-fix-windowsinclude.patch
+        0004-dependencies.patch
+        0005-fix-nasm.patch
+        0007-fix-lib-naming.patch
+        0013-define-WINVER.patch
+        0020-fix-aarch64-libswscale.patch
+        0024-fix-osx-host-c11.patch
+        0040-ffmpeg-add-av_stream_get_first_dts-for-chromium.patch # Do not remove this patch. It is required by chromium
+        0041-add-const-for-opengl-definition.patch
+        0043-fix-miss-head.patch
+        # FIXME should be only applied if feature "fedora-ffmpeg-free-safe" is used
+        0099-ffmpeg-allow-fdk-aac-free.patch
+)
+
+set(OPTIONS "--enable-pic --disable-doc --enable-debug --enable-runtime-cpudetect --disable-autodetect")
+
+if("fedora-ffmpeg-free-safe" IN_LIST FEATURES)
+    set(FEDORA_FFMPEG_FREE_SAFE ON)
+    file(DOWNLOAD "https://src.fedoraproject.org/rpms/ffmpeg/raw/f41/f/ffmpeg_free_sources" "${CURRENT_BUILDTREES_DIR}/ffmpeg_free_sources")
+    file(DOWNLOAD "https://src.fedoraproject.org/rpms/ffmpeg/raw/f41/f/enable_decoders" "${CURRENT_BUILDTREES_DIR}/enable_decoders")
+    file(DOWNLOAD "https://src.fedoraproject.org/rpms/ffmpeg/raw/f41/f/enable_encoders" "${CURRENT_BUILDTREES_DIR}/enable_encoders")
+
+    file(STRINGS "${CURRENT_BUILDTREES_DIR}/ffmpeg_free_sources" FFMPEG_SOURCE_ALLOW_LIST)
+    file(STRINGS "${CURRENT_BUILDTREES_DIR}/enable_decoders" FFMPEG_ENABLE_DECODERS_DIRTY)
+    file(STRINGS "${CURRENT_BUILDTREES_DIR}/enable_encoders" FFMPEG_ENABLE_ENCODERS_DIRTY)
+
+    set(FFMPEG_ENABLE_DECODERS "")
+    set(FFMPEG_ENABLE_ENCODERS "")
+
+    foreach(decoder_dirty IN LISTS FFMPEG_ENABLE_DECODERS_DIRTY)
+        if(decoder_dirty MATCHES "^([^#]+)#?.*$")
+            string(STRIP "${CMAKE_MATCH_1}" decoder)
+            list(APPEND FFMPEG_ENABLE_DECODERS "${decoder}")
+        endif()
+    endforeach()
+    foreach(encoder_dirty IN LISTS FFMPEG_ENABLE_ENCODERS_DIRTY)
+        if(encoder_dirty MATCHES "^([^#]+)#?.*$")
+            string(STRIP "${CMAKE_MATCH_1}" encoder)
+            list(APPEND FFMPEG_ENABLE_ENCODERS "${encoder}")
+        endif()
+    endforeach()
+
+    # Manually whitelisting some low risk sources
```

we try to avoid the whitelist blackist pair, all colors a beautiful. 
```suggestion
    # Manually allow some low risk sources, they include the allowed "libavutil/file_open.c"
```
Are these files used? Can you improve the comment accordingly?

---

I'm struggling to see the value of this comment, but I will prepend it to the first list element, as a middle ground. This list is meant to be something we can extend with specific sources, if we were to drive review on specific codecs, for future needs.

---

Ah Ok, than add this info as comment.